examples/structured_data/deep_neural_decision_forests.py

```diff
@@ -0,0 +1,454 @@
+"""
+Title: Classification with Neural Decision Forests
+Author: [Khalid Salama](https://www.linkedin.com/in/khalid-salama-24403144/)
+Date created: 2021/01/15
+Last modified: 2021/01/15
+Description: Structured data classification with Deep Neural Decision Forests.
```

Comment:
Suggestion:

- Use words from the author's summary/conclusion from the paper to help write more useful descriptions (descriptions shouldn't be just longer versions of titles IMO in this case). Help the users figure out why they should spend their time learning about this technique by scanning the first few paragraphs. The authors have already done the work.

```suggestion
Description: How to model and train stochastic, differentiable decision trees as classifiers for end-to-end learning in deep convolutional networks.
```

Reply:
Thanks for the suggestion. I adapted your description but changed `deep convolutional networks` to `deep learning neural networks`, as we apply the model to structured data, rather images, thus we don't use convolutional networks.